certbot-nginx/certbot_nginx/tls_sni_01.py

```diff
@@ -52,15 +52,9 @@ def perform(self):
             self.configurator.config.tls_sni_01_port)
 
         for achall in self.achalls:
-            vhost = self.configurator.choose_vhost(achall.domain)
-            if vhost is None:
-                logger.error(
-                    "No nginx vhost exists with server_name matching: %s. "
-                    "Please specify server_names in the Nginx config.",
-                    achall.domain)
-                return None
-
-            if vhost.addrs:
+            vhost = self.configurator.choose_vhost(achall.domain, raise_if_no_match=False)
```

Comment:
This is a change in behavior. Previously, when no matching server block was found we'd exit because we didn't know what addresses to use when setting up the challenge. Now we try performing the challenge using all IPv4 interfaces and port 443. This is probably fine, however, I'd add a `logger.info` message explaining this so users have a better idea what may have gone wrong if they fail the challenge. (If you're curious why I think this should be a `logger.info` message rather than using a higher logging level, see #5139).

Also, Apache attempts to ask the user which vhost to use when it encounters this case. If it's unable to ask the user because Certbot is being run non-interactively, it does the same thing as what we're doing here. It certainly doesn't have to be in this PR, but it might not be a bad idea to do the same thing here in the future.

Reply:
Yes, I explictly wanted it to have this behavior. Happy to add an `info` message. I think asking in this case is wrong, because of my assumption that the majority of times we'll hit this is in the "only default is set up" case.